internal/scheduler/legacyscheduler.go

```diff
@@ -533,15 +709,23 @@ func (sched *LegacyScheduler[T]) String() string {
 	return sb.String()
 }
 
-func NewLegacyScheduler[T LegacySchedulerJob](
+func NewLegacyScheduler(
 	ctx context.Context,
 	constraints SchedulingConstraints,
 	config configuration.SchedulingConfig,
 	nodeDb *NodeDb,
-	jobRepository SchedulerJobRepository[T],
+	jobIteratorsByQueue map[string]JobIterator,
 	priorityFactorByQueue map[string]float64,
 	initialResourcesByQueueAndPriority map[string]schedulerobjects.QuantityByPriorityAndResourceType,
-) (*LegacyScheduler[T], error) {
+) (*LegacyScheduler, error) {
+	if len(jobIteratorsByQueue) != len(priorityFactorByQueue) {
```

Comment:
I'd be tempted to combine `jobIteratorsByQueue` and `priorityFactorByQueue` to a `map[string][Queue]` where `Queue` is:
```
 Struct{
    jobIterator JobIterator
    priorityFactor float64    
 }
  ```
That way this function doesn't have to check that they are consistent.  Moreover I'm 99% sure we generate these at the same time (if we don't we have a problem!) so it'll be fewer objects to pass around.

Reply:
I agree. Fixed.